app/controllers/api/v1/metadata_schemas_controller.rb

```diff
@@ -0,0 +1,125 @@
+module Api
+  module V1
+    class MetadataSchemasController < ApiBaseController
+      include Swagger::Blocks
+      before_action :set_available_schemas, only: :index
+      before_action :set_schema_filename, only: :load_schema
+      before_action :set_schema_filepath, only: :load_schema
+      before_action :set_response_type, only: :load_schema
+
+      SCHEMAS_BASE_DIR = Rails.root.join('lib', 'assets', 'metadata_schemas')
+
+      swagger_path '/metadata_schemas' do
+        operation :get do
+          key :tags, [
+              'MetadataSchemas'
+          ]
+          key :summary, 'List all available metadata schemas'
+          key :description, 'Returns a list of all available metadata schemas, by project & version'
+          key :operationId, 'metadata_schemas_path'
+          response 200 do
+            key :description, 'list of available metadata convention schemas'
+          end
+        end
+      end
+
+      def index
+        render json: @schemas
+      end
+
+      swagger_path '/metadata_schemas/{project_name}/{version}/{schema_format}' do
+        operation :get do
+          key :tags, [
+              'MetadataSchemas'
+          ]
+          key :summary, 'Load a metadata convention schema file'
+          key :description, 'Returns a schema definition file for the requested metadata convention, in JSON or TSV format'
+          key :operationId, 'metadata_schemas_load_schema_path'
+          parameter do
+            key :name, :project_name
+            key :in, :path
+            key :description, 'Project name of metadata convention'
+            key :required, true
+            key :enum, %w(alexandria_convention)
+            key :type, :string
+          end
+          parameter do
+            key :name, :version
+            key :in, :path
+            key :description, 'Version of requested convention'
+            key :required, true
+            key :enum, %w(latest 2.0.0 1.1.5 1.1.4 1.1.3)
```

Comment:
This validation using `enum` seems like it would not suffice for multiple projects.

Let's say we release a new convention for a second project, "Foo Metadata Convention 1.0.0", as is foreseeable given we have a URL path segment for `{project_name}` .  Line 43 becomes `key :enum, %w(alexandria_convention foo)` and line 51 becomes `key :enum, %w(latest 2.0.0 1.1.5 1.1.4 1.1.3 1.0.0)`.  Then `/metadata_schemas/foo/1.1.3/json` would pass URL validation -- but it should fail.  Same for `/metadata_schemas/alexandria_convention/1.0.0/json`. 

Perhaps something like a namespaced `enum` in line 51 would address that.  I don't know if that's possible.

I don't think this lurking issue needs to be fixed now, but it seems foreseeable if we add a `project` beyond `alexandria_convention`.

Reply:
So the `enums` are just for Swagger, and not part of any validation server-side.  And since we only have the one project, it works fine for now.  But you are correct in that if we do get another project that has different version numbers, something different will need to be done in order to make those dropdowns consistent.  I'm not aware of anything in Swagger that would support this, but I can check.

Reply:
03f64f7